fix(withdraw): remove Peanut username support from recipient input

FacuBozzi opened the pull request:
Update recipient input validation to reject Peanut usernames for withdrawals and adjust placeholder text accordingly. This change ensures users can only enter valid addresses (banks) or ENS names when withdrawing funds.

FacuBozzi committed:
fix(withdraw): remove Peanut username support from recipient input

Update recipient input validation to reject Peanut usernames for withdrawals and adjust placeholder text accordingly. This change ensures users can only enter valid addresses (banks) or ENS names when withdrawing funds.

FacuBozzi committed:
style: Apply prettier formatting

kushagrasarathe commented on src/components/Global/GeneralRecipientInput/index.tsx:
```diff
@@ -58,6 +58,14 @@ const GeneralRecipientInput = ({
                     const validation = await validateAndResolveRecipient(trimmedInput)
+
+                    // Only accept ENS accounts, reject usernames
+                    if (validation.recipientType.toLowerCase() === 'username') {
```
@FacuBozzi wrong approach, recipient input is used in other parts of the app, and we want usernames to be enabled on those other flows, example direct request, you can try requesting funds in guest flow using `/request/<username>` and enter where you want to request funds. So disabling it directly at root of the component is a bad choice. 

Also the error msg imo seems too long, was this taken from the mockups?
 
<img width="312" alt="Screenshot 2025-06-10 at 12 00 40 PM" src="https://github.com/user-attachments/assets/4fe1dbfe-354a-4c8b-bf4e-84ccddd9acf5" />

FacuBozzi replied:
you're totally right. I started the approach by adding a prop to this specific input/component. Don't know why I changed it. On it.

kushagrasarathe reviewed (changes requested):
@FacuBozzi blocking, please check comment

<img width="306" alt="Screenshot 2025-06-10 at 12 01 21 PM" src="https://github.com/user-attachments/assets/fd8ff31a-799a-45be-a317-c91c7044371b" />

FacuBozzi committed:
feat(components): add prop level validation for withdrawals

add tailwind-colors utility for color management
update GeneralRecipientInput label
iconFillColor prop to AvatarWithBadge and related components

FacuBozzi committed:
Merge branch 'feat/withdraw-ens' of github.com:peanutprotocol/peanut-ui into feat/withdraw-ens

FacuBozzi committed:
fixed error of iconFillColor trying to replace the background color

FacuBozzi committed:
style: Apply prettier formatting

FacuBozzi committed:
withdrawal fixed at prop level (previous commit wasnt isolated and incorrectly also fixed other issues)

FacuBozzi committed:
fixed PFP colors

FacuBozzi committed:
Merge branch 'feat/withdraw-ens' of github.com:peanutprotocol/peanut-ui into feat/withdraw-ens

FacuBozzi committed:
style: Apply prettier formatting

kushagrasarathe commented on src/utils/tailwind-colors.ts:
blocking: why introduce a new utils file? we already have colors defined in `tailwind.config.js`

kushagrasarathe commented on src/components/Global/GeneralRecipientInput/index.tsx:
```diff
@@ -58,6 +60,13 @@ const GeneralRecipientInput = ({
                     const validation = await validateAndResolveRecipient(trimmedInput)
+
+                    // Only accept ENS accounts, reject usernames
+                    if (isWithdrawal && validation.recipientType.toLowerCase() === 'username') {
```
nice

kushagrasarathe commented on src/components/Profile/AvatarWithBadge.tsx:
blocking: changes done in this file are un-necessary, its affecting the transaction card colors, without these changes, it works fine without these changes

<img width="277" alt="image" src="https://github.com/user-attachments/assets/6fa0fa15-54ba-438c-96b5-5b582835e79a" />

kushagrasarathe reviewed (changes requested):
please check comments, blocking

FacuBozzi committed:
refactor(components): remove unused iconBackgroundColor prop and related utils

clean up unused prop across multiple components and delete obsolete tailwind-colors utility

kushagrasarathe reviewed (approved):
noice, lgtm
